stan/math/prim/fun/unit_vector_constrain.hpp

```diff
@@ -52,14 +49,32 @@ template <typename T1, typename T2, require_eigen_col_vector_t<T1>* = nullptr,
 inline plain_type_t<T1> unit_vector_constrain(const T1& y, T2& lp) {
   using std::sqrt;
   check_nonzero_size("unit_vector_constrain", "y", y);
-  return make_holder(
-      [](const auto& y_ref, auto& lp) {
-        value_type_t<T1> SN = dot_self(y_ref);
-        check_positive_finite("unit_vector_constrain", "norm", SN);
-        lp -= 0.5 * SN;
-        return y_ref / sqrt(SN);
-      },
-      to_ref(y), lp);
+  auto&& y_ref = to_ref(y);
+  value_type_t<T1> SN = dot_self(y_ref);
+  check_positive_finite("unit_vector_constrain", "norm", SN);
+  lp -= 0.5 * SN;
+  return y_ref.array() / sqrt(SN);
+}
+
+/**
+ * Return the unit length vector corresponding to the free vector y.
+ * See https://en.wikipedia.org/wiki/N-sphere#Generating_random_points
```

Comment:
* same comments as other doc on return, etc.
* refer to our reference manual, which in turn has references into the literature